software/base/src/main/java/brooklyn/entity/basic/SoftwareProcessImpl.java

```diff
@@ -456,6 +456,9 @@ protected final void callStartHooks() {}
                 Object value = getConfig(k);
                 if (value instanceof Integer){
                     ports.add((Integer)value);
+                } else if (PortRange.class.isAssignableFrom(value.getClass())) {
```

Comment:
452 checks the class of the key's type, then checks assignability of the value from getConfig. If you use `entityspec.configure(Attributes.HTTP_PORT, "9999+")` then it will be picked up on line 452, however if you use `entityspec.configure("http.port"), which is effectively what happens when you put "http.port" in the YAML, then it creates a new `ConfigKey<Object>` which is not picked up on line 452